behavioral/memento.cr

```diff
@@ -0,0 +1,55 @@
+# The memento pattern is a design pattern that permits the current state
+# of an object to be stored without breaking the rules of encapsulation.
+# The originating object can be modified as required but can be restored
+# to the saved state at any time.
+
+# The class whose state can be saved with a memento
+class Match
+  def initialize(fighter1, fighter2, rounds, time)
+    @state = State.new({fighter1, fighter2}, rounds, time)
+  end
+
+  def saveMemento
+    @state
+  end
+
+  def restoreMemento(memento : Match::State)
+    @state = memento
```

Comment:
shouldn't this be `save_memento` and `restore_memento` ?

Reply:
Yeah, definitely. It was a camelCase-esque lapsus from my part. Fixing it as soon as I have a PC available :+1: